src/Form.elm

```diff
@@ -233,15 +288,23 @@ update validation msg (F model) =
                         |> Maybe.map Tree.asList
                         |> Maybe.withDefault []
 
+                filterChangedFields =
+                    Set.filter (not << String.startsWith listName)
```

Comment:
I think this can cause an issue because when you remove an item from a list, the items "after" it take that spot. Since that happens, the item previously at the spot directly after now has the same index/path as the removed item. So all of the `isChanged` type properties actually need to be shifted. I think this is currently a bug in the package.